plugins/source.go

```diff
@@ -33,7 +35,10 @@ type SourcePlugin struct {
 
 type SourceOption func(*SourcePlugin)
 
-const minGoRoutines = 5
+const (
+	minGoRoutines        = 5
+	defaultMaxGoRoutines = 500000
```

Comment:
```suggestion
	defaultMaxGoRoutines = 500000
```

Does this mean that by default we spin up `500000` go routines? Also what's the reason for having a hard `min`? Shouldn't we have only `GoRoutinesCount` and ensure it's not < 0 (0 meaning default value)

Reply:
Yeah I guess you are right - it was from older piece of code where we had an issue with 1-4 go routines.